### DISCUSSION TITLE
[Suggestion] Parameters for the skyblock/auctions endpoint

### DISCUSSION THREAD

**Discussion by IamMusavaRibica** [HypixelDev/PublicAPI#320] [Ideas]
What if one, for example, does not want to send about 50 requests to the auctions endpoint when they don't want most of the returned data?
I suggest to implement parameters for the auctions, for example:
1. category - weapons, armor, accessories, consumables, blocks, misc - Which part of the auctions to return (depending on how it is in the in-game Auction House). To avoid breaking current running applications, if the category parameter is not provided, then everything would be returned
2. type - ah, bin - Whether to return auctions, BINs or both if not provided

This would save a lot of requests and potentially processing power for parsing that data and make applications run faster

**Comment by puppy0cam** [comment]
The API does minimal processing of data, this is unlikely to happen.

**Comment by IamMusavaRibica** [comment]
> The API does minimal processing of data, this is unlikely to happen.

So I'm guessing that the in-game auctions are sorted on the client side, then?

**Comment by AnotherZane** [comment]
> > The API does minimal processing of data, this is unlikely to happen.
> 
> So I'm guessing that the in-game auctions are sorted on the client side, then?

Minimal processing for public requests, not internally.

However, looking at #295 it seems that this endpoint doesn't require authentication anymore so hopefully, they might add something like this.
